master/internal/trial.go

```diff
@@ -1122,6 +1126,10 @@ func (t *trial) terminated(ctx *actor.Context) {
 
 	terminationSent := t.TerminationSent
 
+	if err := t.db.CompleteTrialRun(t.id, t.RunID); err != nil {
```

Comment:
Probably should add a reset operation to restarting a trial. Otherwise, there might be trials that don't have `end_time`

Reply:
can you clarify what you mean?

Reply:
Sorry I was confusing. This `terminated` function might not be called at all if a trial actor fails and exits. I was trying to think of one case that causes trial not to continue after restarting the master.

Reply:
> is if there are any cases that might fail the trial (if the trial actor exits) when it should just restart a trial run.

ah yeah.. hmm. i suppose if the actor panics we could end up with open ended runs which is bad. i'll add a cleanup query to end all open runs in actor.PostStop.